tests/test_template_types.py

```diff
@@ -817,30 +879,48 @@ def test_subject_line_gets_applied_to_correct_template_types():
         LetterPreviewTemplate,
         LetterImageTemplate,
     ]:
-        assert issubclass(cls, WithSubjectTemplate)
+        assert issubclass(cls, SubjectMixin)
     for cls in [
         SMSMessageTemplate,
         SMSPreviewTemplate,
     ]:
-        assert not issubclass(cls, WithSubjectTemplate)
+        assert not issubclass(cls, SubjectMixin)
 
 
-def test_subject_line_gets_replaced():
-    template = WithSubjectTemplate({"content": '', 'subject': '((name))'})
+@pytest.mark.parametrize('template_class, template_type', (
+    (EmailPreviewTemplate, 'email'),
+    (LetterPreviewTemplate, 'letter'),
+))
+def test_subject_line_gets_replaced(template_class, template_type):
+    template = template_class({
+        "content": '', 'template_type': template_type, 'subject': '((name))'
+    })
     assert template.subject == Markup("<span class='placeholder'>((name))</span>")
     template.values = {'name': 'Jo'}
     assert template.subject == 'Jo'
 
 
+@pytest.mark.parametrize('template_class, template_type', (
+    (HTMLEmailTemplate, 'email'),
+    (LetterPrintTemplate, 'letter'),
+))
 @pytest.mark.parametrize("content, values, expected_count", [
     ("Content with ((placeholder))", {"placeholder": "something extra"}, 28),
     ("Content with ((placeholder))", {"placeholder": ""}, 12),
     ("Just content", {}, 12),
     ("((placeholder))  ", {"placeholder": "  "}, 0),
     ("  ", {}, 0),
 ])
-def test_WithSubjectTemplate_character_count(content, values, expected_count):
-    template = WithSubjectTemplate({"content": content, 'subject': 'Hi'})
+def test_WithSubjectTemplate_character_count(
```

Comment:
Should we rename this to get rid of `WithSubjectTemplate` which no longer exists

Reply:
Renamed in c679a44